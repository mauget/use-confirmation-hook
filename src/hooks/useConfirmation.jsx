import React, { useCallback, useState } from 'react';
import ConfirmationModal from '../components/ConfirmationModal';

export default function useConfirmation({
  handleClose,
  headerText,
  bodyText,
}) {
  const [show, setShow] = useState(false);

  const onOpen = () => {
    setShow(true);
  };

  const handleCloseResponse = (response) => {
    handleClose(response);
    setShow(false);
  };

  const Dialog = useCallback(
    () => (
      <ConfirmationModal
        handleClose={handleCloseResponse}
        headerText={headerText}
        show={show}
      >
        {bodyText}
      </ConfirmationModal>
    ),
    [show],
  );

  return { Dialog, onOpen };
}
