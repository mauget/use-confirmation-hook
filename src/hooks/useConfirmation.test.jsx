import React from 'react';
import { shallow } from 'enzyme';
import {
  describe, expect, it, jest,
} from '@jest/globals';
import useConfirmation from './useConfirmation';

const handleCloseMock = jest.fn();

const TestFake = () => {
  const { Dialog, onOpen } = useConfirmation({
    handleClose: handleCloseMock,
    headerText: 'Confirmation Demo',
    bodyText: 'Delete a phantom item?',
  });

  return (
    <>
      <button type="button" onClick={() => onOpen()}>Launch Confirmation Popup</button>
      <Dialog />
    </>
  );
};

describe('useConfirmation', () => {
  it('should execute', () => {
    handleCloseMock.mockReset();

    const wrapper = shallow(<TestFake>Stuff</TestFake>);
    expect(wrapper).toMatchSnapshot();
  });
});
