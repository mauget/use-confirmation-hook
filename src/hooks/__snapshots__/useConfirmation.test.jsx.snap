// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useConfirmation should execute 1`] = `ShallowWrapper {}`;
